docs/appendices/release-notes/unreleased.rst

```diff
@@ -104,3 +104,6 @@ Fixes
 
 - Fixed an issue that could generate duplicate data on ``COPY FROM``  while
   some internal retries were happening due to I/O errors e.g. socket timeouts.
+
+- Fixed an issue that caused ignoring ``routing.allocation.*.{attribute}``
```

Comment:
not sure about phrasing - this issue happens one on partitioned tables, when ALTER TABLE is applied to the whole table, `ONLY` is **not** used and `PARTITION ("partkey"=val)` is **not** used.